src/integrators/sdirk_integrators.jl

```diff
@@ -390,11 +407,8 @@ end
   end
 
   @. z = u - uprev
-  iter = 0
-  κ = cache.κ
-  tol = cache.tol
-
-  iter += 1
+  iter = 1
+  @. u = uprev + z
```

Comment:
I thought so in the beginning as well, but then I got confused because of lines such as https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/pull/141/files/660aa9c90743d54f2aa6c22468ea561ba3480cf9?diff=split#diff-57e331c13262e0ef1f111ed17cda714fL39 where exactly this recalculation of `u` (which might be different to the old `u` because of rounding errors, I guess?) is done before the function evaluation - but also it was only present in the not in-place method... So in the end I decided to do these recalculations in every method. But if it's not needed, I'll just remove it everywhere.